opengrok-indexer/src/test/java/org/opengrok/indexer/history/PerforceRepositoryTest.java

```diff
@@ -105,4 +106,25 @@ public void testHistoryAndAnnotations() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testProtectFilename() throws Exception {
+        PerforceRepository instance = new PerforceRepository();
+        ArrayList<SimpleImmutableEntry<String,String>>   testmap = new ArrayList<SimpleImmutableEntry<String,String>>() {
+            {
+                add(new SimpleImmutableEntry<>("Testfile 34", "Testfile 34"));
+                add(new SimpleImmutableEntry<>("Test%52", "Test%2552"));
+                add(new SimpleImmutableEntry<>("Test*4+2", "Test%2A4+2"));
+                add(new SimpleImmutableEntry<>("Test@", "Test%40"));
+                add(new SimpleImmutableEntry<>("@seventeen", "%40seventeen"));
+                add(new SimpleImmutableEntry<>("upNdown(", "upNdown("));
+                add(new SimpleImmutableEntry<>("tst#99", "tst%2399"));
+            }
+        };
+        for (SimpleImmutableEntry<String,String> ent : testmap) {
+            String prot = instance.protectPerforceFilename(ent.getKey());
+            assertEquals("Improper protected filename, "+prot+" != "+ent.getValue(), ent.getValue(), prot);
+        }
+//        instance.update();
```

Comment:
?

Reply:
removed.  I think I didn't know why it was there.  And, I've since wondered if I should bother having an instance in this test, since it's testing a static method anyway.  Left a comment to that affect in the updated commit.